debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -408,6 +414,13 @@ protected void handleGtidEvent(Event event) {
         String gtid = gtidEvent.getGtid();
         gtidSet.add(gtid);
         source.startGtid(gtid, gtidSet.toString()); // rather than use the client's GTID set
+        ignoreDmlEventByGtidSource = false;
+        if (gtidDmlSourceFilter != null && gtid != null) {
+            String uuid = gtid.trim().substring(0, gtid.indexOf(":"));
+            if (!gtidDmlSourceFilter.test(uuid)) {
```

Comment:
It would probably be better to use a `Set<String>` so that this is faster. Up until now, the filter was not used that frequently. When we create the filter in the `MySqlTaskContext` constructor, we could actually see if the include and exclude config values contain comma-separated UUIDs (rather than regex patterns). I can make that change if you prefer, or I can describe how I might go about doing it so you can give it a try.